File: Numbers/Pyramid/pyramid.js
function pyramid(n) {
    const arr = Array(n * 2 - 1).fill(' ');
    const midLength = Math.floor(arr.length / 2);
    for (let i = midLength, j = midLength; i >= 0; i--, j++) {
        arr[i] = '#';
        arr[j] = '#';
        console.log(`'${arr.join('')}'`);
    }
}

console.log(pyramid(2));
console.log(pyramid(3));
console.log(pyramid(4));
console.log(pyramid(5));
console.log(pyramid(6));

File: Numbers/Pyramid/pyramid.py

from functools import reduce

def pyramid(n):
    arr = [" "] * (n * 2 - 1)
    midLength = len(arr) / 2
    for i, j in zip(range(midLength, -1, -1), range(midLength, len(arr) + 1)):
        arr[i] = '#'
        arr[j] = '#'
        s = reduce(lambda a, b: a + b, arr)
        print(s)

pyramid(1)
pyramid(2)
pyramid(3)
pyramid(4)
pyramid(5)

